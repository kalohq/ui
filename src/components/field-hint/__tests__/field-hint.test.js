/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import FieldHint from '../field-hint';
import Icon from '../../icon';

describe('FieldHint', () => {
  const create = props => <FieldHint {...props} />;

  test('should render shallow component ok', () => {
    const element = renderer.create(<FieldHint />).toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should render with a custom icon', () => {
    const element = create({
      icon: 'archive',
    });

    const result = shallow(element);
    expect(result.find(Icon).length).toBe(1);
  });
});
