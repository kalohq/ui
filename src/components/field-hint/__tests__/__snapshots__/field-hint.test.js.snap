// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldHint should render shallow component ok 1`] = `
<span
  className="e1q9f2xp3 css-5hnrki"
  style={Object {}}
>
  <span
    className="em7k1990 css-9387rp"
    color="navy600"
    onClick={undefined}
    size="extra-small"
    style={Object {}}
  />
</span>
`;
